src/bokeh/core/property/combined.py

```diff
@@ -0,0 +1,53 @@
+#-----------------------------------------------------------------------------
+# Copyright (c) 2012 - 2022, Anaconda, Inc., and Bokeh Contributors.
+# All rights reserved.
+#
+# The full license is in the file LICENSE.txt, distributed with this software.
+#-----------------------------------------------------------------------------
+""" Provide useful aliases of groups of types.
+
+"""
+
+#-----------------------------------------------------------------------------
+# Boilerplate
+#-----------------------------------------------------------------------------
+from __future__ import annotations
+
+import logging # isort:skip
+log = logging.getLogger(__name__)
+
+#-----------------------------------------------------------------------------
+# Imports
+#-----------------------------------------------------------------------------
+
+# Bokeh imports
+from .datetime import Datetime
+from .either import Either
+from .factors import Factor
+from .primitive import Float
+
+#-----------------------------------------------------------------------------
+# Globals and constants
+#-----------------------------------------------------------------------------
+
+__all__ = (
+    'CoordinateLike',
+)
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+CoordinateLike = Either(Float, Datetime, Factor)
```

Comment:
@mattpap Every property links to docs for the property type. This change will possibly break the docs build due to a bad reference, or else leave a missing link for the property type in the reference guide. I don't think that having a property on a model be somehow "not public" is acceptable. Please explain why we can't publicly commit to the name `CoordinateLike` at this point, even if the implementation or internal location is changed or refined in the future? 

More broadly, I also don't understand why the tension between string values and field names, which has existed in several places previously for over ten years, is a concent at all, *in this PR*. Long-established usage of suppling `value` or `field` should still be applicable, in the (uncommon) case that explicit differentiation is necessary.

Reply:
I can't and I don't want to join this discussion, but I am kindly asking for a decision / an advice how to make the final changes, so that this PR can be closed.

Reply:
@mosc9575 we appreciate your patience. @mattpap  and I will resolve any difference separately outside this PR. Until then let's maintain the status quo of built-in models having public, linked property types. Please revert (1) and (2) add then I will merge this PR.

Reply:
@bryevdv Thanks for your fast reply. I try to revert today.